### PR TITLE
fix: add linting to CI + testing-library linting plugin

### PR DESCRIPTION
**Pull request by DrewHoo**

### CODE REVIEW

File: src/controls/ObjectControl.test.tsx [hunk 1]
```diff
@@ -36,8 +36,8 @@ describe("ObjectControl", () => {
           uischema: objectUISchemaWithRule,
         })
 
-        expect(screen.getByText("Name")).not.toBeNull()
```
Comment (NathanFarmer) [inline src/controls/ObjectControl.test.tsx:39]: This change fails tests for some reason. 🍿

**Review by NathanFarmer — CHANGES_REQUESTED** [review]

File: src/controls/ObjectControl.test.tsx (hunk 1 above)
Comment (DrewHoo) [reply src/controls/ObjectControl.test.tsx:39]: It probably needs to be a findBy. Will try that, but is this specific behavior something you disagree with or are you ok with the tests check passing as evidence this is a reasonable change?

Comment (NathanFarmer) [reply src/controls/ObjectControl.test.tsx:39]: No, I just wanted to save this so I could once and for all figure out how this testing framework works.

Comment (DrewHoo) [reply src/controls/ObjectControl.test.tsx:39]: ah, I think it's actually not this change, but a change to cleanup, perhaps that is causing this issue

**Review by NathanFarmer — APPROVED** [review]

**Pull request merged**